examples/routing_example/public/services.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { CoreStart } from 'kibana/public';
+import {
+  RANDOM_NUMBER_ROUTE_PATH,
+  RANDOM_NUMBER_BETWEEN_ROUTE_PATH,
+  POST_MESSAGE_ROUTE_PATH,
+  INTERNAL_GET_MESSAGE_BY_ID_ROUTE,
+} from '../common';
+
+export interface Services {
+  fetchRandomNumber: () => Promise<number | Error>;
+  fetchRandomNumberBetween: (max: number) => Promise<number | Error>;
+  postMessage: (message: string, id: string) => Promise<undefined | Error>;
+  getMessageById: (id: string) => Promise<string | Error>;
+  addSuccessToast: (message: string) => void;
+}
+
+export function getServices(core: CoreStart): Services {
+  return {
+    addSuccessToast: (message: string) => core.notifications.toasts.addSuccess(message),
+    fetchRandomNumber: async () => {
+      try {
+        const response = await core.http.fetch<{ randomNumber: number }>(RANDOM_NUMBER_ROUTE_PATH);
+        return response.randomNumber;
+      } catch (e) {
+        return e;
+      }
```

Comment:
should we remove the `catch(e) throw e` blocks? Or do we want them to show that `fetch` throws in case of code >= 400 ?

Reply:
I wanted consumers to be forced to handle the possibility of an error, which is explicit when the return is `Response | Error` not just `Response`.  

But, I don't know, I'm also torn because then consumers need to do the `isError(response)` check which isn't that nice either.

Reply:
Oh sorry, I misread that for a `catch/throw` instead of `catch/return`.

FYI, platform is going to open a discussion issue on this subject (should core APIs throw errors or return Either<success, error>-ish monads)